Orm/Xtensive.Orm/Modelling/Comparison/Hints/DeleteDataHint.cs

```diff
@@ -17,20 +17,46 @@ namespace Xtensive.Modelling.Comparison.Hints
   [Serializable]
   public class DeleteDataHint : DataHint
   {
+    [Flags]
+    internal enum DeleteDataHintInfo : byte
+    {
+      None = 0,
+      PostCopy = 1,
+      TableMovement = 2,
+      All = PostCopy | TableMovement
+    }
+
+    private readonly DeleteDataHintInfo info = DeleteDataHintInfo.None;
+
     /// <summary>
-    /// Gets or sets a value indicating whether deletion must be performed after completion of copy data hint processing.
+    /// Gets a value indicating whether deletion must be performed after completion of copy data hint processing.
     /// Normally this flag is used to remove records related to types moved to other hierarchies -
     /// these records are still necessary during upgrade to be copied, but must be removed on its
     /// completion.
     /// </summary>
-    public bool PostCopy { get; private set; }
+    public bool PostCopy => info.HasFlag(DeleteDataHintInfo.PostCopy);
+
+    /// <summary>
+    /// Gets a value indicating whether cause of data deletion is due to table have changed its owner type.
+    /// </summary>
+    public bool DueToTableOwnerChange => info.HasFlag(DeleteDataHintInfo.TableMovement);
```

Comment:
What is the resulting IL code? I suspect the `HasFlag` call may cause boxing.
Instead, we can have just a bitwise operation
```csharp
public bool DueToTableOwnerChange => (info & DeleteDataHintInfo.TableMovement) > 0;
```

Reply:
The name would be `IsOwnerChangeCleanup`

Reply:
> What is the resulting IL code? I suspect the `HasFlag` call may cause boxing. Instead, we can have just a bitwise operation
> 
> ```cs
> public bool DueToTableOwnerChange => (info & DeleteDataHintInfo.TableMovement) > 0;
> ```

Yes, it is. I'll change it